jobs/taar_dynamo.py

```diff
@@ -0,0 +1,499 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"""
+This module replicates the scala script over at
+
+https://github.com/mozilla/telemetry-batch-view/blob/1c544f65ad2852703883fe31a9fba38c39e75698/src/main/scala/com/mozilla/telemetry/views/HBaseAddonRecommenderView.scala
+
+This should be invoked with something like this:
+
+spark-submit \
+    --master=spark://ec2-52-32-39-246.us-west-2.compute.amazonaws.com taar_dynamo.py \
+    --date=20180218  \
+    --region=us-west-2  \
+    --table=taar_addon_data_20180206 \
+    --prod-iam-role=arn:aws:iam::361527076523:role/taar-write-dynamodb-from-dev
+
+"""
+
+from datetime import date
+from datetime import timedelta
+from datetime import datetime
+import hashlib
+import os
+from pprint import pprint
+from pyspark import SparkConf
+from pyspark.sql import SparkSession
+from pyspark.sql import Window
+from pyspark.sql.functions import desc, row_number
+import click
+import dateutil.parser
+import json
+import botocore
+import boto3
+from botocore.exceptions import ClientError
+from boto3.dynamodb.types import Binary as DynamoBinary
+import logging
+import time
+import zlib
+
+PATCH_DAYS = timedelta(days=2)
+
+# We use the os and threading modules to generate a spark worker
+# specific identity:w
+
+MAX_RECORDS = 200
+EMPTY_TUPLE = (0, 0, [], [])
+
+BOTO_CFG = botocore.config.Config(retries={"max_attempts": 16})
```

Comment:
make max_attempts configurable?

Reply:
I don't think there's any advantage to making this configurable.  

The primary problem we see in practice is cross cloud network throttling.  To work around that - I've put in an exponential back-off over here: https://github.com/mozilla/telemetry-airflow/pull/815/files#diff-3f86d29de1b5bb05b05e589750cae922R221